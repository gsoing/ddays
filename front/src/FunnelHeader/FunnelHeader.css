
.FunnelHeader {
  font-family: 'RenaultLife';
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 60px;
  padding: 8px 16px;
  width: 100%;
  background-color: white;
  color: #1f2532;
  border-bottom: 1px solid #e6e6ec;
  text-align: left;
  
}
@media (min-width: 670px) {
  .FunnelHeader {
    height: 70px;
  }
}

.FunnelHeader__title {
  display: block;
  flex: 1;
  margin: 0;
  font-size: 1.6rem;
  /* font-family: themefont('brand'); */
  text-transform: uppercase;

    font-size: 1.8rem;

}

@media (min-width: 670px) {
  .FunnelHeader__title {
    font-size: 1.8rem;
  }
}

.FunnelHeader__homelink {
  display: block;
  width: 145px;
}
