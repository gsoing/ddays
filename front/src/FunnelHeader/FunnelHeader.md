Example :

    <FunnelHeader title="Welcome to paradise" titleAsH1 homeLinkUrl="http://www.google.fr" homeLinkTitle="Bar" />
