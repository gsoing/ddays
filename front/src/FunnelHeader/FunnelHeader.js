import React from 'react';

import './FunnelHeader.css';
import mainLogo from '../img/mainLogo.svg';

const FunnelHeader = ({ title }) => (
  <header className="FunnelHeader">
    <span className="FunnelHeader__title">{title}</span>
    <span className="FunnelHeader__homelink">
      <img className="FunnelHeader__brand" src={mainLogo} />
    </span>
  </header>
);

export default FunnelHeader;
