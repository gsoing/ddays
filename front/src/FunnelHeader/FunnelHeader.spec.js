import React from 'react';
import renderer from 'react-test-renderer';
import FunnelHeader from './FunnelHeader';

jest.mock('../../../common/Link/Link', () => jest.fn(({ children }) => <div>Link component:{children}</div>));

describe('funnel header component', () => {
  it('should render correctly with default props', () => {
    const component = renderer.create(
      <FunnelHeader />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with forced H1 title', () => {
    const component = renderer.create(
      <FunnelHeader title="Foo" titleAsH1 homeLinkUrl="http://test" homeLinkTitle="Bar" />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with forced span title', () => {
    const component = renderer.create(
      <FunnelHeader title="Foo" titleAsH1={false} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
