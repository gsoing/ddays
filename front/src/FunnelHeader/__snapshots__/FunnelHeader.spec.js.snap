// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`funnel header component should render correctly with default props 1`] = `
<header
  className="FunnelHeader"
>
  <span
    className="FunnelHeader__title"
  >
    
  </span>
  <div>
    Link component:
    <img
      alt={undefined}
      className="FunnelHeader__brand"
      src="test-file-stub"
    />
  </div>
</header>
`;

exports[`funnel header component should render correctly with forced H1 title 1`] = `
<header
  className="FunnelHeader"
>
  <h1
    className="FunnelHeader__title"
  >
    Foo
  </h1>
  <div>
    Link component:
    <img
      alt="Bar"
      className="FunnelHeader__brand"
      src="test-file-stub"
    />
  </div>
</header>
`;

exports[`funnel header component should render correctly with forced span title 1`] = `
<header
  className="FunnelHeader"
>
  <span
    className="FunnelHeader__title"
  >
    Foo
  </span>
  <div>
    Link component:
    <img
      alt={undefined}
      className="FunnelHeader__brand"
      src="test-file-stub"
    />
  </div>
</header>
`;
